Add coefficient for matrix assembly

DavidSCN opened the pull request:
... as discussed. The change is rather simple as deal.II has predefined functionality to handle the coefficient.

Follow-up question is where the function should store the assembled matrix. At the moment, the content is always stored in the `stationary_system_matrix_`, i.e., calling `setup_system(coeff)` multiple times will override previous content of the matrix. If we want to keep the content of the matrix we either need to copy it on the python side before re-assembling the matrix for a different coefficient or we need to handle things different on the deal.II side. @sdrave any opinion on this?

Regarding the handling of Dirichlet boundary conditions in deal.II, their [documentation](https://dealii.org/developer/doxygen/deal.II/namespaceMatrixTools.html) (excerpt of the section 'Global elimination') writes: 

>  For this reason, we perform a very simple line balancing by not setting the main diagonal entry to unity, but rather to the value it had before deleting this line, or to the first nonzero main diagonal entry if it is zero for some reason. Of course we have to change the right hand side appropriately.

Thus, the column and row of the constrained DoF have only one non-zero entry  (the value of the unconstrained system) and the coupling towards other DoFs is resolved using Gauss elimination.

sdrave commented:
Hi @DavidSCN, thanks for the changes. Unfortunately, we seem to have misunderstood each other regarding the parametrization. In your current implementation the diffusivity is globally constant. Scaling it by a factor will not yield any interesting. One simple parametrization would be to let the parameter be the diffusivity in a fixed subdomain and choose constant value outside this subdomain.

> Follow-up question is where the function should store the assembled matrix. At the moment, the content is always stored in the `stationary_system_matrix_`, i.e., calling `setup_system(coeff)` multiple times will override previous content of the matrix. If we want to keep the content of the matrix we either need to copy it on the python side before re-assembling the matrix for a different coefficient or we need to handle things different on the deal.II side. @sdrave any opinion on this?

I would let `setup_system` create a new `SparseMatrix` and return it.

DavidSCN commented:
> Unfortunately, we seem to have misunderstood each other regarding the parametrization. In your current implementation the diffusivity is globally constant. Scaling it by a factor will not yield any interesting. One simple parametrization would be to let the parameter be the diffusivity in a fixed subdomain and choose constant value outside this subdomain.

Alright, fixed it. I thought you want to handle the coefficient split on the python side using global matrices. I added now four parameters in order to specify the diffusion coefficient: two parameters for the domain split in x and y and two coefficient values. This leads of course to discontinuous coefficients.

> I would let setup_system create a new SparseMatrix and return it

Done as well. Ownership is now passed to the python side. Before merging here, I need to clean up a bit and remove the previously used getter for the system matrix in order to prevent misuse.
